core/headerchain.go

```diff
@@ -493,7 +511,7 @@ func (hc *HeaderChain) SetHead(head uint64, delFn DeleteCallback) {
 	}
 	hc.currentHeaderHash = hc.CurrentHeader().Hash()
 
-	rawdb.WriteHeadHeaderHash(hc.chainDb, hc.currentHeaderHash)
```

Comment:
why move this above? the hc.currentHeaderHash was updated before the write.

Reply:
Since the cache field shall be updated only after the successful db write. Else this is not an atomic operation, and will cause data discrepancy in memory cache and storage.

Reply:
I see. cool.